### ISSUE TITLE
Feature Request: Android/iOS Apps

### ISSUE THREAD

**Issue by balupton** [marktext/marktext#639]
### Describe your feature request

Would love to see an iOS App for Mark Text

### What problem does this feature solve? [optional]

Editing markdown files on iOS, for the purpose of editing blog posts while in bed, or on the run.

**Comment by balupton** [comment]
Perhaps https://github.com/marktext/muya will help here.

**Comment by Jocs** [comment]
Yes, we have plans to launch the mobile version of the markdown editor, but it will take some time, at least the desktop version needs to be stable.

**Comment by ymjonline** [comment]
it would be awesome but first the desktop app needs to be able to manage notes in Notebooks (based on folder structure ?) andTags (based on YAML front matter header)

Then sync (dropbox-like) and view / edit from mobile.

I would not expect the mobile version to be in-line editing but at least a view/edit toggle.

Then and only then will the frustrating quest for a Evernote-like replacement will be quenched.

thanks !

**Comment by balupton** [comment]
Evernote and Mark Text are different markets.

Evernote is a clipping product, as you use the web you can clip highlights of content, or resources themselves, with their contextual information, for the purchase of a context aware knowledge repository.

Tools like Bear Notes, Mark Text, Simple Note, are not in that space. They seem more just your general repository of written notes, so the market of journals, of which dairies a subset.

**Comment by Jocs** [comment]
This issue need help, I you have any idea or want to help develop IOS/Android version, pls comment in this issue.

**Comment by balupton** [comment]
Regarding sync, encryption, and collab. I now work at @textileio which offers free and open-source tooling to accomplishing those three things, it aims to be a decentralised iCloud.

We'd be happy to team up to assist in development of those features when you are ready to add them.

**Comment by brainchild0** [comment]
It would be wonderful to have a stable and usable application like  Mark Text on a mobile platform.

It also would  be wonderful to have Mark Text stable and usable on the desktop. Mostly, Mark Text is stable and usable, but there are still questions that need to be answered and problems that need to be resolved.  Until then, it seems too early to direct attention toward supporting a different device class. 

As Mark Text matures, along with frameworks for embedded web applications, it will be clearer how to design or to create a mobile version.

**Comment by tauren** [comment]
Could mobile apps be built as PWAs using the [File System Access API](https://wicg.github.io/file-system-access/)? Might save time by not having to write native apps. But I haven't tried to access local files, so I don't know for sure if this will work.

And maybe integrate with [SyncThing](https://syncthing.net/). I've found it to be a great open source solution for syncing between cross-platform devices without need for a centralized server. On Android, [syncthing-fork](https://github.com/Catfriend1/syncthing-android) has been a robust client for me, but I never tried the original since I read it wasn't as well maintained as the fork.

**Comment by brainchild0** [comment]
@tauren: I think these tools have less relevance than you may suggest to the current issue.

**Comment by Apreche** [comment]
I have recently been trying out many Markdown editors, and was very happy to find Marktext. I wish I had found it sooner. However, the inability to use it on iOS is really a pain. Either a native iOS app or a web app will be fine, it doesn't matter which one for me. I just want to be able to use MarkText on my iPad the same as I do on my desktop. The fact that this issue has been open for three years doesn't give me much hope.

**Comment by brainchild0** [comment]
The application has not seen active development recently, and a mobile application would  be essentially a different application supported by the same project. Porting the current application to a mobile target seems to be not realistic.

It is understandable how a mobile application would be appealing, but as stated already, it may not be feasible now.

It would be more practical to start or to support a project dedicated to a mobile application than to integrate mobile support into the current project. If anyone reading has the will and resources to do so, then please consider it!

**Comment by Apreche** [comment]
> The application has not seen active development recently, and a mobile application would  be essentially a different application supported by the same project. Porting the current application to a mobile target seems to be not realistic.
> 
> 
> 
> It is understandable how a mobile application would be appealing, but as stated already, it may not be feasible now.
> 
> 
> 
> It would be more practical to start or to support a project dedicated to a mobile application than to integrate mobile support into the current project. If anyone reading has the will and resources to do so, then please consider it!

The only reason I don't do it myself is because I only own iOS devices, but no MacOS devices. If Apple released XCode for iPad, I'd absolutely make an attempt.

**Comment by Ovilia** [comment]
For mobile devices, I would suggest using cross-platform solutions because it could save a lot of efforts.

I would personally suggest using [ionic](https://ionicframework.com/) because this project is written in JavaScript and ionic uses TypeScript so it's easier to transfer and maintain. And ionic is really an awesome platform which I used for many projects. If you decide to work on the mobile thing, maybe I could offer some help.

Right now, what we need most is one or more guys to play as owner[s] who is familiar with this project and more importantly, have time and enthusiasm for working or mobile apps.

**Comment by brainchild0** [comment]
I think   many assume,  because Ionic, like Electron, is based on web technologies,  that it is trivial to develop a mobile version of an application built on Electron. In fact, development under Electron, targeting multiple major desktop platforms with essentially no platform-specific development, is possible for two reasons. First, desktop platforms have now converged to essentially the same functionality, each borrowing the best features from the others. Second, abstraction layers, such as Node, have reached a level of sophistication that overcomes the historic   tradeoff between flexibility versus portability. 

Mobile platforms are different from desktop ones in important ways, not only in underlying architecture, but also user interfaces. Mobile applications must present a screen layout, and accommodate an input method, different from those on desktop and laptop machines. Even writing screen layouts suitable for mobile displays would be a strain on the extremely limited development capacity for this project.

If anyone wishes to commit development efforts to a mobile version of the application, then I would not discourage it in the least. My own view carries two parts. First, a robust mobile version of the application would be more realistic once a robust desktop version is available, and gains more widespread recognition. Second, until the essential code base becomes more mature, it is unrealistic to expect core development resources to move into the mobile space.

**Comment by Jocs** [comment]
In fact, there are plans to develop iOS and Android versions, which may be based on Flutter, but the specific product logic and technology have not been investigated yet. Where are the files(data) stored? Whether to support cloud contributions, etc. are all issues that need to be considered, so currently it is only in the idea stage.

**Comment by brainchild0** [comment]
I am glad there are plans for a mobile version. I would love to use an editor like Mark Text on my mobile device. However, I also want to see  improvements in stability and usability on the desktop. My only concern about mobile development is it might move attention away from  the existing desktop application.

**Comment by brainchild0** [comment]
> Whether to support cloud stores, etc. are all issues that need to be considered, so currently it is only in the idea stage.

You might consider existing projects, such as [Markor](https://github.com/gsantner/markor), for ideas.

I think you're right that questions such as storage are more complicated on a mobile device than a desktop. 

A desktop application relies mostly on the user supplying the location and name of a file to restore or to save. On a mobile platform, the user expects integration with some more complicated storage back end. For this reason, it becomes more important to consider   integration with other applications and services. 

It may be easier to test certain critical concepts on a desktop version before becoming excited about making it work on the mobile targets.

**Comment by joshiojas** [comment]
> Perhaps https://github.com/marktext/muya will help here.

How does muya help? does it have an IOS and android app?

**Comment by Jocs** [comment]
>How does muya help? does it have an IOS and android app?

You can develop an iOS/Android app with muya by Flutter, RN etc. We have no plans yet, and the focus is still on the stability of the desktop and muya editor

**Comment by vip20** [comment]
I am using [Joplin](https://joplinapp.org/) for this. It supports all major platforms. But I am eagerly waiting for Marktext on mobile devices as the UI is amazing.

**Comment by TickDracy** [comment]
@Jocs Do you know if this [app](https://www.lessfun.com/app/marktext/) is based on Marktext, even though it has the same name?

**Comment by rjvisser** [comment]
@Jocs is a release for mobile platforms still in the works?
I'd be very interested in this as well for the use case I'm looking into: versioned requirements management captured as use-cases in markdown format managed through Git. 
So the synchronization between the mobile and desktop space could be managed via Git - meaning that the mobile application would only need to offer text editing, similar to what the desktop application is offering.
My suggestion would be to start with that as a minimal viable solution and see whether there are any sync-features to be added later.

Having the same workflow on both the Desktop and the mobile platforms would make it easier for people to adopt the workflow

**Comment by oilop9000** [comment]
[termux GUI](https://wiki.termux.com/wiki/Graphical_Environment) ??

**Comment by Jocs** [comment]
Thank you for the interest in a mobile version of MarkText. An iOS/Android app would be a fundamentally different product from our Electron desktop application — it requires a different runtime, a touch-first UI, platform-specific file-access APIs, and its own distribution story. After six years on the tracker there has been no concrete implementation work, and the project's current capacity is focused on the desktop editor and the @muyajs/core engine rewrite.

We are closing this as out-of-scope for the desktop project. If anyone in the community wants to build a mobile Markdown editor that reuses the muya editor engine, the engine is published as the @muyajs/core npm package and can be embedded in a web view on mobile. We would be happy to answer engine API questions in a dedicated discussion. Thank you for your enthusiasm for the project.
